ProjectHive/ProjectHive.Services.NotificationsAPI/Services/WebSocketHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Newtonsoft.Json;
+using ProjectHive.Services.NotificationsAPI.Dto;
+using ProjectHive.Services.NotificationsAPI.Models;
+using System.Net.WebSockets;
+using System.Text;
+
+namespace ProjectHive.Services.NotificationsAPI.Services
+{
+    public class WebSocketHandler :IWebSocketHandler
+    {
+        private readonly List<WebSocket> _sockets = new List<WebSocket>();
+
+        public async Task SendNotificationAsync(NotificationDto notification)
+        {
+            var message = JsonConvert.SerializeObject(notification);
+            var buffer = Encoding.UTF8.GetBytes(message);
+            var segment = new ArraySegment<byte>(buffer);
+
+            foreach (var socket in _sockets)
+            {
+                if (socket.State == WebSocketState.Open)
+                {
+                    await socket.SendAsync(segment, WebSocketMessageType.Text, true, CancellationToken.None);
+                }
+            }
+        }
+
+        public void AddSocket(WebSocket socket)
+        {
+            _sockets.Add(socket);
+        }
+    }
+}
```

Comment:
Also, please implement a mechanism to remove closed or faulty WebSocket connections from the `_sockets` list. 
`RemoveSocket()` and so on

Reply:
And ensure please that WebSocket connections are properly disposed of to free up resources.
I do not remember properly but I suppose a disposal method should be implemented there as well

Reply:
Also please do not forget logs for some important places, it will help you to identify issues much more quickly.